packages/model-viewer/src/features/scene-graph/api.ts

```diff
@@ -49,6 +49,32 @@ export declare interface Model {
    * @returns the first material to whose name matches `name`
    */
   getMaterialByName(name: string): Material|null;
+
+  /**
+   * Creates a new material variant from an existing material.
```

Comment:
Might be nice to explain when you'll get `null`.

Reply:
Done

Reply:
Can you be a  little more specific than "when it doesn't succeed"? I would not generally expect this method to fail without reading its implementation. What restrictions are on the input to ensure it succeeds?

Reply:
I reworked this to remove the fail case, so now it always succeeds.